pdg/src/main.rs

```diff
@@ -175,21 +175,134 @@ fn main() -> eyre::Result<()> {
 
 #[cfg(test)]
 mod tests {
-    use std::{env, path::Path, process::Command};
+    use std::{
+        env,
+        fmt::Display,
+        path::{Path, PathBuf},
+        process::Command,
+    };
 
-    use color_eyre::eyre;
+    use color_eyre::eyre::{self, ensure, eyre, Context};
+
+    use crate::{Pdg, ToPrint};
+
+    pub enum Profile {
+        Debug,
+        Release,
+        // Will be used by future test code.
+        #[allow(dead_code)]
+        Other(String),
+    }
+
+    impl Profile {
+        pub fn name(&self) -> &str {
+            use Profile::*;
+            match self {
+                Debug => "dev",
+                Release => "release",
+                Other(other) => other.as_str(),
+            }
+        }
+
+        pub fn dir_name(&self) -> &str {
+            use Profile::*;
+            match self {
+                Debug => "debug",
+                Release => "release",
+                Other(other) => other.as_str(),
+            }
+        }
+    }
+
+    impl From<String> for Profile {
+        fn from(profile: String) -> Self {
+            use Profile::*;
+            match profile.as_str() {
+                "debug" => Debug,
+                "release" => Release,
+                _ => Other(profile),
+            }
+        }
+    }
+
+    impl Profile {
+        pub fn current() -> eyre::Result<Self> {
+            let profile =
+                env::var("PROFILE").wrap_err(eyre!("should be set by `build.rs` from `cargo`"))?;
+            Ok(profile.into())
+        }
+    }
+
+    pub fn repo_dir() -> eyre::Result<PathBuf> {
+        let crate_dir = env::var("CARGO_MANIFEST_DIR")?;
+        let repo_dir = Path::new(&crate_dir)
+            .parent()
+            .ok_or_else(|| eyre!("`$CARGO_MANIFEST_DIR` should have a parent"))?;
+        Ok(repo_dir.to_owned())
+    }
+
+    fn pdg_snapshot_inner(
```

Comment:
This function does several things without explaining beforehand or inline what it's trying to do, and its name isn't enough to explain what exactly it should do. Let's have a doc comment on it along the lines of the comment in `pdg.sh`.

Reply:
Okay, I'll add that.  Should it be on `pdg_snapshot_inner` or on `pdg_snapshot`, as `pdg_snapshot` is meant to be the one you call (the inner one is so the long function doesn't have to be generic)?

Reply:
As `pdg_snapshot_inner` is where the nontrivial logic is implemented, the logic should be explained there. A high-level doc comment on the wrapper too wouldn't hurt, of course.

Reply:
Fixed in https://github.com/immunant/c2rust/pull/636/commits/84bb03335cf512509a3eaeba5fc71c3a2ebcd57d.